Google Calendar: Recurrent event has TypeError

### The problem

After upgrade of core, recurrent events from google calendar stopped showing in calendar

### What version of Home Assistant Core has the issue?

core-2022.11.4

### What was the last working version of Home Assistant Core?

_No response_

### What type of installation are you running?

Home Assistant Supervised

### Integration causing the issue

Google Calendar

### Link to integration documentation on our website

https://www.home-assistant.io/integrations/google

### Diagnostics information

_No response_

### Example YAML snippet

_No response_

### Anything in the logs that might be useful for us?

```txt
Traceback (most recent call last):
File "/usr/local/lib/python3.10/site-packages/aiohttp/web_protocol.py", line 435, in _handle_request
resp = await request_handler(request)
File "/usr/local/lib/python3.10/site-packages/aiohttp/web_app.py", line 504, in _handle
resp = await handler(request)
File "/usr/local/lib/python3.10/site-packages/aiohttp/web_middlewares.py", line 117, in impl
return await handler(request)
File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 60, in security_filter_middleware
return await handler(request)
File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 100, in forwarded_middleware
return await handler(request)
File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 28, in request_context_middleware
return await handler(request)
File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 82, in ban_middleware
return await handler(request)
File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 236, in auth_middleware
return await handler(request)
File "/usr/src/homeassistant/homeassistant/components/http/view.py", line 136, in handle
result = await result
File "/usr/src/homeassistant/homeassistant/components/calendar/__init__.py", line 264, in get
calendar_event_list = await entity.async_get_events(
File "/usr/src/homeassistant/homeassistant/components/google/calendar.py", line 453, in async_get_events
return [
File "/usr/src/homeassistant/homeassistant/components/google/calendar.py", line 453, in <listcomp>
return [
File "/usr/local/lib/python3.10/site-packages/ical/iter.py", line 234, in overlapping
for item in self._iterable:
File "/usr/local/lib/python3.10/site-packages/ical/iter.py", line 143, in __next__
self._make_heap()
File "/usr/local/lib/python3.10/site-packages/ical/iter.py", line 133, in _make_heap
next_item = next(iterator)
File "/usr/local/lib/python3.10/site-packages/ical/iter.py", line 113, in __iter__
for dtvalue in self._recur:
File "/usr/local/lib/python3.10/site-packages/gcal_sync/timeline.py", line 106, in __iter__
for value in self._func:
File "/usr/local/lib/python3.10/site-packages/gcal_sync/timeline.py", line 92, in __iter__
for value in self._dt_iter:
File "/usr/local/lib/python3.10/site-packages/ical/iter.py", line 113, in __iter__
for dtvalue in self._recur:
File "/usr/local/lib/python3.10/site-packages/dateutil/rrule.py", line 1396, in _iter
heapq.heapify(rlist)
File "/usr/local/lib/python3.10/site-packages/dateutil/rrule.py", line 1338, in __lt__
return self.dt < other.dt
TypeError: can't compare datetime.datetime to datetime.date
```


### Additional information

_No response_

Thank you, I am able to reproduce the event with the data you've provided, with an event like this:
```
        event = Event.parse_obj(
            {
                "id": "event-id",
                "summary": "Summary",
                "start": {
                    "date": "2022-04-19",
                },
                "end": {
                    "date": "2022-04-20",
                },
                "recurrence": [
                    "RDATE;VALUE=DATE:20221115,20221120,20221125,20221205",
                    "RRULE:FREQ=DAILY;UNTIL=20221030;INTERVAL=5"
                ],
            }
        )
...
tests/test_timeline.py:916: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../ical/ical/iter.py:234: in overlapping
    for item in self._iterable:
../ical/ical/iter.py:143: in __next__
    self._make_heap()
../ical/ical/iter.py:133: in _make_heap
    next_item = next(iterator)
../ical/ical/iter.py:113: in __iter__
    for dtvalue in self._recur:
gcal_sync/timeline.py:106: in __iter__
    for value in self._func:
gcal_sync/timeline.py:92: in __iter__
    for value in self._dt_iter:
../ical/ical/iter.py:113: in __iter__
    for dtvalue in self._recur:
venv/lib/python3.9/site-packages/dateutil/rrule.py:1396: in _iter
    heapq.heapify(rlist)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <dateutil.rrule.rruleset._genitem object at 0x7f484cc22b50>
other = <dateutil.rrule.rruleset._genitem object at 0x7f484cc221c0>

    def __lt__(self, other):
>       return self.dt < other.dt
E       TypeError: can't compare datetime.datetime to datetime.date

venv/lib/python3.9/site-packages/dateutil/rrule.py:1338: TypeError
```

I'll work on a fix.

--- a/homeassistant/components/google/manifest.json
+++ b/homeassistant/components/google/manifest.json
@@ -4,7 +4,7 @@
   "config_flow": true,
   "dependencies": ["application_credentials"],
   "documentation": "https://www.home-assistant.io/integrations/calendar.google/",
-  "requirements": ["gcal-sync==4.0.2", "oauth2client==4.1.3"],
+  "requirements": ["gcal-sync==4.0.3", "oauth2client==4.1.3"],
   "codeowners": ["@allenporter"],
   "iot_class": "cloud_polling",
   "loggers": ["googleapiclient"]

--- a/requirements_all.txt
+++ b/requirements_all.txt
@@ -738,7 +738,7 @@ gTTS==2.2.4
 garages-amsterdam==3.0.0
 
 # homeassistant.components.google
-gcal-sync==4.0.2
+gcal-sync==4.0.3
 
 # homeassistant.components.geniushub
 geniushub-client==0.6.30

--- a/requirements_test_all.txt
+++ b/requirements_test_all.txt
@@ -554,7 +554,7 @@ gTTS==2.2.4
 garages-amsterdam==3.0.0
 
 # homeassistant.components.google
-gcal-sync==4.0.2
+gcal-sync==4.0.3
 
 # homeassistant.components.geocaching
 geocachingapi==0.2.1

